presto-main/src/test/java/com/facebook/presto/operator/BenchmarkWindowOperator.java

```diff
@@ -0,0 +1,330 @@
+
+            return pagesBuilder.build();
+        }
+
+        private RowPagesBuilder buildPages(int currentPartitionIdentifier, List<Type> typesArray)
+        {
+            int groupIdentifier = 100;
+            RowPagesBuilder rowPagesBuilder = RowPagesBuilder.rowPagesBuilder(false, ImmutableList.of(0), typesArray);
+
+            for (int i = 0; i < TOTAL_PAGES; i++) {
+                BlockBuilder firstColumnBlockBuilder = BIGINT.createBlockBuilder(null, ROWS_PER_PAGE);
+                BlockBuilder secondColumnBlockBuilder = BIGINT.createBlockBuilder(null, ROWS_PER_PAGE);
+                int currentNumberOfRowsInPartition = 0;
+                int numberOfPartitionsInCurrentGroup = 0;
+                int currentGroupIdentifier = groupIdentifier++;
+
+                for (int j = 0; j < ROWS_PER_PAGE; j++) {
+                    if (currentNumberOfRowsInPartition == rowsPerPartition) {
+                        ++currentPartitionIdentifier;
+                        ++numberOfPartitionsInCurrentGroup;
+                        currentNumberOfRowsInPartition = 0;
+                    }
+
+                    if (numberOfPartitionsInCurrentGroup == partitionsPerGroup) {
+                        numberOfPartitionsInCurrentGroup = 0;
+                        currentGroupIdentifier = groupIdentifier++;
+                    }
+
+                    firstColumnBlockBuilder.writeLong(currentGroupIdentifier);
+                    secondColumnBlockBuilder.writeLong(currentPartitionIdentifier);
+                    ++currentNumberOfRowsInPartition;
+                }
+
+                rowPagesBuilder.addBlocksPage(
+                        firstColumnBlockBuilder.build(),
+                        secondColumnBlockBuilder.build(),
+                        createLongSequenceBlock(0, ROWS_PER_PAGE),
+                        createLongSequenceBlock(0, ROWS_PER_PAGE));
+            }
+
+            return rowPagesBuilder;
+        }
+
+        public TaskContext createTaskContext()
+        {
+            return TestingTaskContext.createTaskContext(executor, scheduledExecutor, TEST_SESSION, new DataSize(2, GIGABYTE));
+        }
+
+        public OperatorFactory getOperatorFactory()
+        {
+            return operatorFactory;
+        }
+
+        public List<Page> getPages()
+        {
+            return pages;
+        }
+    }
+
+    @Benchmark
+    public List<Page> benchmark(BenchmarkWindowOperator.Context context)
+    {
+        DriverContext driverContext = context.createTaskContext().addPipelineContext(0, true, true).addDriverContext();
+        Operator operator = context.getOperatorFactory().createOperator(driverContext);
+
+        Iterator<Page> input = context.getPages().iterator();
+        ImmutableList.Builder<Page> outputPages = ImmutableList.builder();
+
+        boolean finishing = false;
+        for (int loops = 0; !operator.isFinished() && loops < 1_000_000; loops++) {
```

Comment:
Why the 1M limit? It may mask the effect of changes that affect how many loops are required to complete the operation.

Reply:
@nezihyigitbasi I have opened an issue for refactoring benchmarks, please see above.